### ISSUE TITLE
Issue while running wrf-chem model

### ISSUE THREAD

**Issue by burhangis** [anikfal/atmospheric_science#17]
I ran wrf-chem model after executing this command (./prep_chem_sources_RADM_WRF_FIM_.exe) i got these files
![github](https://github.com/user-attachments/assets/db8978a6-f4d9-4e42-ac80-77b7dead1233)

but it is giving an error on ./real.exe command which is as follows
"FATAL CALLED FROM FILE:  <stdin>  LINE:     424
error opening met_em.d01.2019-09-04_12:00:00.nc for input; bad date in namelist or file not in directory"

**Comment by anikfal** [comment]
Are you new in running the WRF model? If so, for the moment, I recommend to run the model with no chemistry, so you will be familiar with the model structure. After that you can try running WRF/Chem with emissions data.

Anyway, the error is related to the WRF model itself and is self-explanatory. The date/time does not match with `namelist.wps`.

**Comment by burhangis** [comment]
yes i am new in WRF and also i successfully ran the model without checmistry module by using ERA-5 datasets and it worked fine

**Comment by anikfal** [comment]
I think I have already shared my namelists. You can use them for your runs.

**Comment by burhangis** [comment]
ok sure let me check it

**Comment by burhangis** [comment]
Hi
i chcked an issue and it is saying that met em_d01.... file not exist in directory..I did not see any command about ./metgrid in your videos before executing .real.exe in your directory met... files exist....

**Comment by burhangis** [comment]
hi i ran ungrib and metgrid commands by taking era5 dataset after that i am running .real.exe but it giving me an error. Please highlight my mistake
Thanks

![image](https://github.com/user-attachments/assets/c3d178a2-9e2b-4560-837a-d738f89080d0)

**Comment by burhangis** [comment]
pubdata1/Model/New-HPC/


live:.cid.1d869697fb92cee9, 8:17 PM
http://www.linkedin.com/comm/mynetwork/discovery-see-all?usecase=PEOPLE_FOLLOWS&followMember=williamhatheway


LinkedIn Login, Sign in | LinkedIn
http://www.linkedin.com

live:.cid.1d869697fb92cee9, 8:24 PM

bash file.txt

Hi
I am running ./geogrid.exe and ./ungrib.exe command on supercomputer but it giving me error that

/geogrid.exe: error while loading shared libraries: libnetcdff.so.7: cannot open shared object file: No such file or directory
and

/usr/mathlib/libpng/lib/libpng15.so.15: version `PNG15_0' not found (required by ./ungrib.exe)

respectively...in my specific path libnetcdff.so.7 is not available but libnetcdff.so.6 is available and libpng15.so.15 is available in specific path.

I also added the path (export LD_LIBRARY_PATH=/usr/mathlib/libpng/lib/) in bash file

how i can resolve these issues?

**Comment by anikfal** [comment]
Hi.
On an HPC system, you usually need to load the required libraries before running the programs (`geogrid.exe`, `ungrib.exe`, etc.). Loading can be done by `module load netcdf-lib`. You are supposed to replace the appropriate name instead of `netcdf-lib`, representing the NetCDF libraries installed on the HPC system.

**Comment by burhangis** [comment]
i already setup env variables in bash file(url) 
[bash file.txt](https://github.com/user-attachments/files/18230021/bash.file.txt)

**Comment by anikfal** [comment]
I think you might not have included the NetCDF-Fortran libraries in the `NETCDF` installation directory. `libnetcdff.so.7` is related to NetCDF Fortran.

**Comment by burhangis** [comment]
on my HPC i have libnetcdff.so.6.2.1 instead of libnetcdff.so.7 so how i can change it or fix the problem with available library

**Comment by anikfal** [comment]
in the `NETCDF` installation path, try to link the installed library as `libnetcdff.so.7`. It may help to resolve the issue:
`ln -s path/to/libnetcdff.so.6.2.1 $NETCDF/libnetcdff.so.7`

**Comment by burhangis** [comment]
it giving an error of permission denied
![Uploading Screenshot 2024-12-24 185440.png…]()

**Comment by anikfal** [comment]
Try to link both NetCDF C and Fortran libraries in your home directory as the required file names (e.g. `libnetcdff.so.7`) and then:
`export LD_LIBRARY_PATH=/path/to/netcdf_c/lib:/path/to/netcdf_fortran/lib:$LD_LIBRARY_PATH`

**Comment by burhangis** [comment]
Actually on my HPC i have not installed libnetcdff.so.7 which is causing an error
![Screenshot 2024-12-24 193653](https://github.com/user-attachments/assets/6116095a-6656-4d9c-8738-0e7b1c56b037)

**Comment by anikfal** [comment]
> ln -s /public/software/mathlib/miniconda3/pkgs/netcdf-fortran-4.4.5-hea25ff8_1000/lib/libnetcdff.so.6.2.1 ./libnetcdff.so.7

This line is critical. For other files, just simply link them with no new names.

If it does not help, try to install NetCDF libraries locally on you home directory.

**Comment by burhangis** [comment]
hi
i am installing of prep_chem_source utilities but when i ran this command 
make OPT=intel.wrf CHEM=RADM_WRF_FIM

it giving me following error. What is the solution of this error?
![Screenshot 2025-01-15 162814](https://github.com/user-attachments/assets/958e2f75-5ade-417e-a202-13ec73369419)

Thanks

**Comment by anikfal** [comment]
It seems to be an Intel compiler issue. try to use GCC instead.

**Comment by burhangis** [comment]
In my supercomputer i configured wrf-chem with intel compiler when i try with gfortran it giving me this error
![Uploading Screenshot 2025-01-15 164648.png…]()

**Comment by anikfal** [comment]
If your NetCDF libraries are compiled with Intel, then you need to compile prep_chem with Intel as well. I think you need to modify and setup `intel.wrf` correctly.

PS. Please use [code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks). Don't paste screenshot images.

**Comment by burhangis** [comment]
I am already modified intel.wrf due to intel compiler. attaching interl.wrf file


`
# Define make (gnu make works best).

MAKE=/usr/bin/make

# Activate appropriate parts below, comment out others.

# NCAR Graphics.

#---------------------------------------------------------------
# If you are using a standard installation of NCAR Graphics, set:
#       LOADER=ncargf90
# in the machine-dependent sections below
#LIBNCARG=
#---------------------------------------------------------------
# If you are using the NCAR dummy libraries...

NCARG_DIR=$(BASE)
#LIBNCARG=-L$(NCARG_DIR) -lncarg-$(UTILS_VERSION) -lncarg_c-$(UTILS_VERSION) \
#                        -lncarg_gks-$(UTILS_VERSION)
#LIBNCARG=-L$(NCARG_DIR) -lncarg-$(UTILS_VERSION) 
LIBNCARG=$(BASE)/libncarg-$(UTILS_VERSION)-$(OPT).a
#---------------------------------------------------------------
# If you are using a real distribution of NCAR Graphics...
#NCARG_DIR=/usr/local/ncarg-4.3.0/lib
#LIBNCARG=-L$(NCARG_DIR) -lncarg -lncarg_gks -lncarg_c -L/usr/X11R6/lib -lX11 -ldl
#---------------------------------------------------------------

# NETCDF libraries


NETCDF=/public/software/mathlib/netcdf/intel/4.6.1

NETCDF_INC=-I$(NETCDF)/include 
NETCDF_LIBS=-L$(NETCDF)/lib -lnetcdff -lnetcdf

# HDF libraries


HDF5=/public/software/mathlib/hdf5/1.8.20
HDF5_INC=-I$(HDF5)/include
HDF5_LIB=-L$(HDF5)/lib -lhdf5hl_fortran -lhdf5_fortran -lhdf5_hl -lhdf5 -L/public/software/mathlib/zlib/1.2.8/lib -lz -ldl	
	 
# Machine-dependent options.

#-----------------  LINUX Portland Group pgf77/gcc ---------------
CMACH=PC_LINUX1
F_COMP=ifort
C_COMP=gcc
LOADER=ifort
C_LOADER=gcc
LIBS=
MOD_EXT=mod

F_OPTS=-fpp -D$(CHEM) -O2 -convert big_endian
C_OPTS= -O2 
LOADER_OPTS= -O2

#-----------------------------------------------------------------


# If compiling for a single-CPU platform only (without MPI):

#-----------------------------------------------------------------
PAR_LIBS=
PAR_DEFS=
#-----------------------------------------------------------------


# For IBM,HP,SGI,ALPHA use these:
ARCHIVE=ar rs
# For NEC SX-6
#ARCHIVE=sxar rs
# For SUN,CONVEX
#ARCHIVE=ar r'`

**Comment by burhangis** [comment]
My issue has been resolve and i successfully build  prep_chem_sources_RADM_WRF_FIM_.exe file

**Comment by anikfal** [comment]
What was the issue exactly?

**Comment by burhangis** [comment]
The cause of this error is ironically simple: the absence of remark symbol ('!') in front of the 9th line of script src/convert_edgar_to_RELACS_REAC.f90 file

**Comment by anikfal** [comment]
Please copy the line here. And how did you find out that this was the cause of the issue?

**Comment by burhangis** [comment]
Open the file named as **convert_edgar_to_RELACS_REAC.f90** which will under `PREP-CHEM-SRC-1.5\src` folder and add **!** at line 9 like below

!use chem1_list

**Comment by anikfal** [comment]
I guess you must have found the solution from [here](https://forum.mmm.ucar.edu/threads/compiling-prep_chem_sources-1-5-solved.1780/). Happy to see that it is resolved.
